rules/DeadCode/Rector/StmtsAwareInterface/RemoveJustPropertyFetchRector.php

```diff
@@ -147,14 +158,29 @@ private function replaceVariablesWithPropertyFetch(
         // remove assign node
         unset($stmtsAware->stmts[$currentStmtsKey]);
 
+        // avoid crash in \PHPStan\Analyser\MutatingScope::enterAnonymousFunctionWithoutReflection
+        $variableCompatibleFetch = clone $propertyFetch;
+        if (!$variableCompatibleFetch->name instanceof Node\Identifier && !is_string($variableCompatibleFetch->name)) {
+            throw new ShouldNotHappenException();
+        }
+        $variableCompatibleFetch->name = (string) $variableCompatibleFetch->name;
```

Comment:
cast `(string)` is not needed, replace with `Identifier` should be ok.

Reply:
actually, it seems should can be removed, because assign to same data or Identifier.

Reply:
This is the key of the fix - converting `Identifier` to `string`, because PhpStan throws an exception if it gets a name that is not a string

Reply:
This is where it crashed before this fix:
https://github.com/phpstan/phpstan-src/blob/1.8.x/src/Analyser/MutatingScope.php#L2846

Reply:
I tried remove the assignment and it seems working ok:

```diff
-        $variableCompatibleFetch->name = $variableCompatibleFetch->name;
```

Reply:
Looks like with all the additional edge case handling this work-around is not needed anymore, will remove.